src/rpc/misc.cpp

```diff
@@ -36,32 +36,33 @@
 
 static RPCHelpMan validateaddress()
 {
-    return RPCHelpMan{"validateaddress",
-                "\nReturn information about the given bitcoin address.\n",
-                {
-                    {"address", RPCArg::Type::STR, RPCArg::Optional::NO, "The bitcoin address to validate"},
-                },
-                RPCResult{
-                    RPCResult::Type::OBJ, "", "",
+    return RPCHelpMan{
+        "validateaddress",
+        "\nReturn information about the given bitcoin address.\n",
+        {
+            {"address", RPCArg::Type::STR, RPCArg::Optional::NO, "The bitcoin address to validate"},
+        },
+        RPCResult{
+            RPCResult::Type::OBJ, "", "",
+            {
+                {RPCResult::Type::BOOL, "isvalid", "If the address is valid or not"},
+                {RPCResult::Type::STR, "address", /* optional */ true, "The bitcoin address validated"},
+                {RPCResult::Type::STR_HEX, "scriptPubKey", /* optional */ true, "The hex-encoded scriptPubKey generated by the address"},
+                {RPCResult::Type::BOOL, "isscript", /* optional */ true, "If the key is a script"},
+                {RPCResult::Type::BOOL, "iswitness", /* optional */ true, "If the address is a witness address"},
+                {RPCResult::Type::NUM, "witness_version", /* optional */ true, "The version number of the witness program"},
+                {RPCResult::Type::STR_HEX, "witness_program", /* optional */ true, "The hex value of the witness program"},
+                {RPCResult::Type::STR, "error", /* optional */ true, "Error message, if any"},
+                {RPCResult::Type::ARR, "error_locations", /*optional=*/true, "Indices of likely error locations in address, if known (e.g. Bech32 errors)",
```

Comment:
I was just curious about this thing. In this section of code, two ways of commenting the argument as optional are used:
1. `/* optional */`
2. `/*optional=*/`

I have noticed that in this PR, you have used the second way of commenting for all the comments you added and have not changed the preexisting comments.
If the second way of commenting is the right way of the two. Maybe it could be taken up in follow-up PRs to refactor the type 1 comments.

Reply:
Yeah, this is being done in https://github.com/bitcoin/bitcoin/pull/23545

Reply:
> this is being done in #23545

That's great! Just have to wait for a little to resolve PR conflicts with #23545